libvast/src/system/partition.cpp

```diff
@@ -805,6 +805,12 @@ partition_actor::behavior_type passive_partition(
       // We can safely assert that if we have the partition chunk already, all
       // deferred evaluations were taken care of.
       VAST_ASSERT(self->state.deferred_evaluations.empty());
+      // Don't handle queries after we already received an exit message, while
+      // the terminator is running. Since we require every partition to have at
+      // least one indexer, we can use this to check.
```

Comment:
I'm not quite sure if I can understand the comment. Are all `indexers` cleared when entering shutdown? (I did not see a call to `clear` in the EXIT handler.)

Reply:
Yes, correct, they're cleared in the exit handler via `std::exchange(self->state.indexers, {})`.

Reply:
Ah, sure, because everybody who reads this knows that this means "clearing a vector". 🤓

Reply:
Yeah, I think with the benefit of 10 years of hindsight, I would say that `std::move()` should have just been `static_cast<T&&>`, and that the function `std::move()` should have the semantics of `std::exchange()`. But it's too late to change now, so we might as well get used to it :D

Reply:
To be fair, the proposal for `std::take` that essentially is `std::exchange(x, {})` is apparently being rejected because the name of the function is too short. Maybe we'll get it as `co_take` sometime in the future. 🤷‍♂️